pkg/kubelet/config/config.go

```diff
@@ -413,10 +425,19 @@ func podsDifferSemantically(existing, ref *api.Pod) bool {
 
 // checkAndUpdatePod updates existing, and:
 //   * if ref makes a meaningful change, returns needUpdate=true
+//   * if ref makes a meaningful change, and this change is graceful deletion, returns needGracefulDelete=true
 //   * if ref makes no meaningful change, but changes the pod status, returns needReconcile=true
-//   * else return both false
-//   Now, needUpdate and needReconcile should never be both true
-func checkAndUpdatePod(existing, ref *api.Pod) (needUpdate, needReconcile bool) {
+//   * else return all false
+//   Now, needUpdate, needGracefulDelete and needReconcile should never be both true
+func checkAndUpdatePod(existing, ref *api.Pod) (needUpdate, needReconcile, needGracefulDelete bool) {
+	// 1. this is an graceful delete
+	if ref.DeletionTimestamp != nil {
+		existing.DeletionTimestamp = ref.DeletionTimestamp
+		needGracefulDelete = true
+		return
+	}
```

Comment:
nits: I mean, it would be safer to update the config cache even for the graceful deletion.
So combine this block with the `update` part to make sure the config cache is updated. :)